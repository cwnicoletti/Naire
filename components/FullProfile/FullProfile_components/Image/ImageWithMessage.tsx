import React, { useEffect } from "react";
import {
  StyleSheet,
  Platform,
  Image,
  View,
  Dimensions,
  TouchableNativeFeedback,
  TouchableOpacity,
} from "react-native";
import MaskedView from "@react-native-masked-view/masked-view";
import { FontAwesome, FontAwesome5 } from "@expo/vector-icons";
import { LinearGradient } from "expo-linear-gradient";

const ImageWithMessage = (props) => {
  const width = Dimensions.get("window").width;

  let TouchableCmp: any = TouchableOpacity;
  if (Platform.OS === "android") {
    TouchableCmp = TouchableNativeFeedback;
  }

  return (
    <View>
      <Image
        source={{
          uri: props.image,
        }}
        style={{ height: width, width: width, marginRight: 15 }}
      />
      {props.showMessage ? (
        <TouchableCmp
          onPress={() => {
            if (props.setPicture) {
              props.setPicture(props.image);
            }
            if (props.navigation) {
              props.navigation.getParent()?.setOptions({
                tabBarStyle: { display: "none" },
              });
            }
            props.onOpen();
          }}
        >
          <View
            style={{
              position: "absolute",
              backgroundColor: "white",
              justifyContent: "center",
              alignItems: "center",
              alignSelf: "flex-end",
              right: 20,
              bottom: 20,
              height: 50,
              width: 50,
              shadowOffset: {
                width: 0,
                height: 0,
              },
              shadowColor: "black",
              shadowOpacity: 0.1,
              shadowRadius: 6,
              borderRadius: 50 / 2,
            }}
          >
            <MaskedView
              style={{ flex: 1 }}
              maskElement={
                <View
                  style={{
                    flex: 1,
                    justifyContent: "center",
                    alignItems: "center",
                  }}
                >
                  <FontAwesome
                    name="envelope-o"
                    size={28}
                    color="black"
                    style={{ position: "absolute" }}
                  />
                  <FontAwesome5
                    name="heart"
                    size={11}
                    color="black"
                    style={{ marginBottom: 5 }}
                  />
                </View>
              }
            >
              <LinearGradient
                colors={["#A700D1", "#434aa8"]}
                style={{
                  flex: 1,
                  height: 50,
                  width: 50,
                }}
              />
            </MaskedView>
          </View>
        </TouchableCmp>
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({});

export default ImageWithMessage;
