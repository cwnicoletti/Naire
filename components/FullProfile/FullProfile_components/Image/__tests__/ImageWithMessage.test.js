import React from 'react';
import renderer from 'react-test-renderer';
import ImageWithMessage from '../ImageWithMessage';

jest.mock('react-native-vector-icons', () => 'Icon');

describe('<ImageWithMessage />', () => {
  it('renders enabled ImageWithMessage', () => {
    const tree = renderer.create(<ImageWithMessage />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders <ImageWithMessage /> with custom props', () => {
    const tree = renderer
      .create(
        <ImageWithMessage
          image={
            'https://res.cloudinary.com/personaluse1234/image/upload/v1643742678/Naire/Genevieve%20Hannelius/nbvmnmbn3333_1_yw6lh7.jpg'
          }
          showMessage={true}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
