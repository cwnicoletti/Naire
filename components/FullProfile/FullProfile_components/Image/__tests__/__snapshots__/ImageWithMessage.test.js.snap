// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageWithMessage /> renders <ImageWithMessage /> with custom props 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": "https://res.cloudinary.com/personaluse1234/image/upload/v1643742678/Naire/Genevieve%20Hannelius/nbvmnmbn3333_1_yw6lh7.jpg",
      }
    }
    style={
      Object {
        "height": 750,
        "marginRight": 15,
        "width": 750,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "backgroundColor": "white",
          "borderRadius": 25,
          "bottom": 20,
          "height": 50,
          "justifyContent": "center",
          "position": "absolute",
          "right": 20,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 6,
          "width": 50,
        }
      }
    >
      <RNCMaskedView
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text />
            <Text />
          </View>
        </View>
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4289134801,
                4282600104,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 50,
              "width": 50,
            }
          }
        />
      </RNCMaskedView>
    </View>
  </View>
</View>
`;

exports[`<ImageWithMessage /> renders enabled ImageWithMessage 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 750,
        "marginRight": 15,
        "width": 750,
      }
    }
  />
</View>
`;
